### ISSUE TITLE
Increase max number of domains per vhost

### ISSUE THREAD

**Issue by fbatschi** [NginxProxyManager/nginx-proxy-manager#2664]
**Is your feature request related to a problem? Please describe.**
Right now, each redirection host is limited to only have 15 domains (coming from the `maxOptions` in the render function of the forms. When you manage a big portfolio of domains this is quite limiting to have to setup identical vhosts, splitting up to 15 domains each, just because of the domain limit.

**Describe the solution you'd like**

Let's Encrypt for instance supports up to 100 SAN in a certificate so I don't see why there should be a limit of 15 domains in a vhost in nginx proxy manager. Increase the limit to 100 - or make the number configurable either via env vars/docker-compose vars or configurable via backend what the limit of domains per vhost should be.

**Comment by phillf** [comment]
Why not use a wildcard instead of individually naming every domain you want to serve? Just do *.example.com.

**Comment by fbatschi** [comment]
its not about subdomains, but many individual 2nd level domain names (i am talking about hundreds of domains example1.com, example2.com, etc). right now i have to artificially split them into vhosts of 15 each.

With subdomains and using wildcards the same question would then arise. i would have to request wildcard Let's Encrypt certificates as well (which only works via dns) or also face the limit of 15 domains per form.

 i am not sure what the thought behind this limit for form input was. is it a UI thing?

**Comment by phillf** [comment]
@fbatschi, Was the original question posed a hypothetical or do you legitimately manage hundreds of domains? If yes, I honestly wonder if nginx proxy manager is the appropriate tool for your use case.

**Comment by fbatschi** [comment]
Yes i actually manage several hundreds of domains that basically just do a simple http redirect. I used to edit nginx vhost configs file directly, but since somebody else should now be able to manage the redirects with a proper GUI, nginx proxy manager seems to be a great choice.
 
I will probably patch that `maxOptions` value myself and use my own docker build for it.

Nevertheless i thought i would submit this feature request, since i don't really see where the motivation for this arbitrary limit of `15` input values for all forms comes from. UX wise it would mean less duplicate settings if that limit would be higher. 

But i also understand, that only very few users would probably be impacted by it.

**Comment by nrgbistro** [comment]
I have a use case for more than 15 subdomain items to one host. I'm using authentik to act as a proxy with authentication, so I want to point any subdomains I want to pass through authentik to the authentik host. Encountered this problem and thread while trying to add a 16th subdomain to authentik. For now I'm creating another entry to the same host, but as @fbatschi said this increases clutter in the UI. I would love the option to configure this seemingly arbitrary limit with a docker ENV.

**Comment by nrgbistro** [comment]
👍

**Comment by phillf** [comment]
👍

**Comment by aktso** [comment]
I found values of a number of maxOptions are already 100 which appear to limit number of selected domain names. Just want to know whether the mentioned issue still exist.

**Comment by fbatschi** [comment]
> I found values of a number of maxOptions are already 100 which appear to limit number of selected domain names. Just want to know whether the mentioned issue still exist.

it seems indeed to be fixed (well, increased to 100 which would be sufficient for me for now). 

closing this now.
